gluten-core/src/main/scala/io/glutenproject/utils/PullOutProjectHelper.scala

```diff
@@ -57,12 +57,13 @@ trait PullOutProjectHelper {
 
   protected def replaceExpressionWithAttribute(
       expr: Expression,
-      projectExprsMap: mutable.HashMap[Expression, NamedExpression]): Expression =
+      projectExprsMap: mutable.HashMap[Expression, NamedExpression],
+      replaceBoundReference: Boolean = true): Expression =
```

Comment:
The default value should be false, and `GenerateExec` set it to true. "replace" means use an `Alias` wrapper.